server/api/image.py

```diff
@@ -147,6 +151,28 @@ def download(self, image, params):
         return file_stream
 
     @describeRoute(
+        Description('Return histograms of image metadata.')
+        .param('filter',
+               'Get the histogram after the results of this filter. ' +
+               'TODO: describe our filter grammar (an AST tree).',
+               required=False)
+        .param('limit', 'Result set size limit. Setting to 0 will create ' +
+               'a histogram using all the matching items (default=0).',
+               required=False, dataType='int')
+        .param('offset', 'Offset into result set (default=0).',
+               required=False, dataType='int')
+        .errorResponse()
+    )
+    @access.public
+    def histogram(self, params):
+        # TODO: ideally we should access images via the image route, but it
```

Comment:
AFAIK, we need the collection to run mapreduce code.

We *do* [prefilter](https://github.com/ImageMarkup/isic-archive/blob/images-gallery-serverside-histograms/server/histogram_utility/__init__.py#L33) the collection before running mapreduce—it's kind of an optimized, all-in-one call.